### PR TITLE
Escape all characters special to markdown.

### PR DESCRIPTION
**Pull request by brondsem**
These commits will escape all characters that markdown handles specially.  As you can see in the test changes, there are a lot of unnecessary escapes (e.g. for periods and hyphens).  But the escaping is necessary in some cases.  This might be better as an option, if some people prefer prettier output over escaping everything.

### CODE REVIEW

**Comment by aaronsw** [comment]
Yeah, I think this will drive most people up the wall -- it should either escape only when necessary or be an option, default off.

**Comment by brondsem** [comment]
I've updated my code to make this an option, off by default.

It'd be great if it could do the extra escaping automatically but only when necessary.  But I didn't know how to determine when that would be.

File: test/normal.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ first issue
   * _**bold italic**_
     * orange
     * apple
+    * yam\\sweet potato
```
Comment (aaronsw) [inline test/normal.md]: I don't think we should be adding tests with actually wrong results -- this is obviously not the correct behavior.

**Comment by brondsem** [comment]
Good point; I've added a commit to revert the changes to the 'normal' test files.

**Pull request merged**